services/kinesis/src/test/java/software/amazon/awssdk/services/kinesis/SubscribeToShardUnmarshallingTest.java

```diff
@@ -101,7 +101,7 @@ public void exceptionWithMessage_UnmarshalledCorrectly() throws Throwable {
             fail("Expected ResourceNotFoundException exception");
         } catch (ResourceNotFoundException e) {
             assertThat(e.requestId()).isEqualTo(REQUEST_ID);
-            assertThat(e.statusCode()).isEqualTo(500);
+            assertThat(e.statusCode()).isEqualTo(400);
```

Comment:
note: this is a bugfix. `ResourceNotFoundException` [doesn't have a fault trait](https://github.com/aws/aws-sdk-java-v2/blob/eeee2389d4d1f0d776ac75e492469dea9c817a0e/services/kinesis/src/main/resources/codegen-resources/service-2.json#L1916-L1926), so it should be 400.